tests/testthat/_snaps/verb-uncount.md

```diff
@@ -8,7 +8,15 @@
       FROM (
         SELECT `x`, `w`, `..dbplyr_row_id`
         FROM `test` AS `LHS`
-        INNER JOIN `dbplyr_table` AS `RHS`
+        INNER JOIN (
+      SELECT CAST(`..dbplyr_row_id` AS INTEGER) AS `..dbplyr_row_id`
+      FROM (
+        SELECT NULL AS `..dbplyr_row_id`
+        WHERE (0 = 1)
+        UNION ALL
+        VALUES (1)
+      ) AS `values_table`
+        ) AS `RHS`
           ON (`RHS`.`..dbplyr_row_id` <= `LHS`.`w`)
```

Comment:
@mgirlich: The indentation is off here. I wonder if we want an `op_df()` that we can insert into the SQL expression tree that would, when rendered, be turned into a CTE?

Reply:
When I implemented `copy_inline()` I knew that in these situations the indentation is off but I thought it would take a least a little bit longer until someone complains 😄
The issue is that the SQL for `copy_inline()` is created directly, so it cannot know about the level where it needs to be rendered.

Even with CTE the indentation is off

```r
dbplyr_uncount(df, w) %>% show_query(cte = TRUE)
```

```sql
WITH `q01` AS (
  SELECT `x`, `w`, `..dbplyr_row_id`
  FROM `test` AS `LHS`
  INNER JOIN (
SELECT CAST(`..dbplyr_row_id` AS INTEGER) AS `..dbplyr_row_id`
FROM (
  SELECT NULL AS `..dbplyr_row_id`
  WHERE (0 = 1)
  UNION ALL
  VALUES (1)
) AS `values_table`
  ) AS `RHS`
    ON (`RHS`.`..dbplyr_row_id` <= `LHS`.`w`)
)
SELECT `x`
FROM `q01`
```

I'll take a look how much effort it is to change this.

I guess in this case we want, if CTEs are activated, the values clause to be rendered in a CTE. I'm not sure whether there are other cases like this in the dbplyr code. But I already thought it might be interesting to have a user facing function for this.

Reply:
With explicitly turning on CTEs the query looks much better after https://github.com/tidyverse/dbplyr/pull/810.
Do you still see the need for an `op_vars()` to turn parts into a CTE?

Reply:
With `op_vars` we would get `WITH q01 AS (SELECT ...) WITH q02 AS SELECT ... INNER JOIN q01 ...` ? That would be a tad more readable.

Does this issue arise whenever a `copy_inline()` is used as a joined or unioned table?

Reply:
Isn't the first example in #810 exactly what you want?

```sql
WITH `q01` AS (
  SELECT CAST(`..dbplyr_row_id` AS INTEGER) AS `..dbplyr_row_id`
  FROM (
    SELECT NULL AS `..dbplyr_row_id`
    WHERE (0 = 1)
    UNION ALL
    VALUES (1)
  ) AS `values_table`
),
`q02` AS (
  SELECT `x`, `w`, `..dbplyr_row_id`
  FROM `test` AS `LHS`
  INNER JOIN `q01` AS `RHS`
    ON (`RHS`.`..dbplyr_row_id` <= `LHS`.`w`)
)
SELECT `x`
FROM `q02`
```